packages/nameguard-react/src/components/ReportIcon/index.tsx

```diff
@@ -63,57 +46,46 @@ export function ReportIcon({
   */
   ...props
 }: ReportShieldProps) {
-  const onClickHandler = (clickHandlerFor: ClickHandlerFor) => {
-    switch (clickHandlerFor) {
-      case ClickHandlerFor.icon:
-        if (onIconClickOverride) {
-          onIconClickOverride(ensName);
-        } else {
-          redirectToViewNameReportURL(ensName);
-        }
-        break;
-      case ClickHandlerFor.tooltip:
-        if (onTooltipClickOverride) {
-          onTooltipClickOverride(ensName);
-        } else {
-          redirectToViewNameReportURL(ensName);
-        }
-        break;
-    }
-  };
+  let icon: React.ReactNode;
+  let tooltipIcon: React.ReactNode;
+  let tooltipTitleClass: string;
+  let tooltipTitle: string;
+  let tooltipSubtitle: string | undefined = undefined;
+  let tooltipMessage: string | undefined = undefined;
+
+  // TODO: please share advice if we should add `role="button"` to each of the `icon` elements defined below? I saw that applied in the old code. Was it right to remove it?
 
   if (hadLoadingError) {
-    return (
-      <ReportUnknownIcon
-        size={size}
-        className="cursor-pointer"
-        onIconClickOverride={(e?: React.MouseEvent) => {
-          if (e) e.stopPropagation();
-          onClickHandler(ClickHandlerFor.icon);
-        }}
-        onTooltipClickOverride={(e?: React.MouseEvent) => {
-          if (e) e.stopPropagation();
-          onClickHandler(ClickHandlerFor.tooltip);
-        }}
+    icon = <RatingUnknownIcon size={size} isInteractive={true} {...props} />;
+    tooltipIcon = <RatingUnknownIcon size={RatingIconSize.small} />;
+    tooltipTitleClass = "font-semibold mb-1 text-white";
+    tooltipTitle = "Error loading report";
+  } else if (!data) {
+    icon = <RatingLoadingIcon size={size} {...props} />; // TODO: add isInteractive={true} prop?
+    tooltipIcon = (
+      <RatingLoadingIcon
+        size={RatingIconSize.small}
+        fill="#dddddd"
+        {...props}
       />
+    ); // TODO: why is it necessary to pass this `fill` prop to make this visible? That's a bad design. It should work properly without needing to pass that.
```

Comment:
Hi @lightwalker-eth, I have already spent an hour on this today and I remember to have invested in the past on this strange need. In fact it is not nice to have to set `fill` in here but curiously this is needed for the Tooltip version of `RatingLoadingIcon`... Maybe it is good to come back to this after merging your PR?

Reply:
@FrancoAguzzi Agreed, I created a separate story to track this goal here: https://app.shortcut.com/ps-web3/story/25745/refine-loading-state-of-ratingicon

Reply:
Added comments to this story so we further define this for future coding! 🙂